File: manager/src/main/java/moe/shizuku/manager/model/ServiceStatus.java
package moe.shizuku.manager.model;

import moe.shizuku.api.ShizukuService;

public class ServiceStatus {

    private int uid;
    private int version;
    private String secontext;

    public ServiceStatus() {
    }

    public boolean isRunning() {
        return ShizukuService.pingBinder();
    }

    public int getUid() {
        return uid;
    }

    public void setUid(int uid) {
        this.uid = uid;
    }

    public int getVersion() {
        return version;
    }

    public void setVersion(int version) {
        this.version = version;
    }

    public String getSEContext() {
        return secontext;
    }

    public void setSEContext(String secontext) {
        this.secontext = secontext;
    }
}

File: manager/src/main/java/moe/shizuku/manager/model/ServiceStatus.kt
package moe.shizuku.manager.model

import rikka.shizuku.Shizuku

data class ServiceStatus(
        val uid: Int = -1,
        val apiVersion: Int = -1,
        val patchVersion: Int = -1,
        val seContext: String? = null,
        val permission: Boolean = false
) {
    val isRunning: Boolean
        get() = uid != -1 && Shizuku.pingBinder()
}